examples/leasing/node_release_script.py

```diff
@@ -0,0 +1,241 @@
+                node_project = node_status[1]
+                node_duration = node_status[2]
+    status_file.close()
+    return (node_project, node_duration)
+
+
+def update_time_for_node(
+        statusfile, node,
+        old_time, new_time
+        ):
+    """Updates the time for a node in file"""
+
+    newline = ''
+    with open(statusfile, 'r') as status_file,\
+            open('/tmp/tempfile', 'w') as output_file:
+        for line in status_file:
+            words = line.split()
+            if node == words[0]:
+                words[2] = str(new_time)
+                newline = ' '.join(words)+"\n"
+                output_file.write(newline)
+            else:
+                output_file.write(line)
+    status_file.close()
+    output_file.close()
+
+    remove(statusfile)
+    move('/tmp/tempfile', statusfile)
+
+
+def update_project_in_status_file(
+        statusfile, node, new_project, old_project
+        ):
+    """Project will be changed in
+    file to match the project in HIL"""
+
+    newline = ''
+    with open(statusfile, 'r') as status_file,\
+            open('/tmp/tempfile', 'w') as output_file:
+        for line in status_file:
+            words = line.split()
+            if node == words[0]:
+                words[1] = new_project
+                newline = ' '.join(words) + "\n"
+                output_file.write(newline)
+            else:
+                output_file.write(line)
+    status_file.close()
+    output_file.close()
+    remove(statusfile)
+    move('/tmp/tempfile', statusfile)
+
+
+def detach_networks_from_node(hil_client, node):
+    """Disconnect all networks from all of the node's NICs
+    get node information and then iterate on the nics"""
+
+    node_info = hil_client.node.show(node)
+
+    for nic in node_info['nics']:
+        # get the port and switch to which the nics are connected to
+        port = nic['port']
+        switch = nic['switch']
+        if port and switch:
+            try:
+                hil_client.port.port_revert(switch, port)
+                print('Removed all networks from node `%s`' % node)
+            except FailedAPICallException:
+                print('Failed to revert port `%s` on node \
+                        `%s` switch `%s`' % (port, node, switch))
+                raise HILClientFailure()
+
+
+def release_from_project(
+        hil_client, statusfile, node, project
+        ):
+    """Updates the file with project name as free and
+    releases it back to free pool"""
+
+    time.sleep(2)
+    # tries 2 times to detach the project because there might be a pending
+    # networking action setup by revert port in the previous step.
+    counter = 2
+    while counter:
+
+        try:
+            hil_client.project.detach(project, node)
+            print('Node `%s` removed from project `%s`' % (node, project))
+            break
+        except FailedAPICallException as ex:
+            if ex.message == 'Node has pending network actions':
```

Comment:
We should probably modify the client library to expose the type of error, rather than client programs to look at the human-readable description.

Reply:
Yeah, I had written this in some other hil client, and at that time I just wanted the client to work rather than dealing with the client library itself. I'll open an issue about it.